server/controllers/articles.js

```diff
@@ -0,0 +1,248 @@
+import database from "../models/index";
+import { articleSchema } from "../validationSchemas/article";
+import { validateData } from "../validations/validateData";
+import { generateUniqueSlug } from "../helpers/generateUniqueSlug";
+import { calculateReadTime } from "../helpers/calculateReadTime";
+import { httpResponse, serverError } from "../helpers/http";
+import { registerUser } from "../tests/fixtures/model/userData";
+
+const Article = database["Articles"];
+
+/**
+ * @description Create a new Article
+ *
+ * @param {object} req HTTP request object
+ * @param {object} res HTTP response object
+ * @returns {object}  Response message object
+ */
+export const createArticle = async (req, res) => {
+  const result = await validateData(req.body, articleSchema);
+  if (result.errorMessages) {
+    return httpResponse(res, {
+      statusCode: 400,
+      message: result.errorMessages
+    });
+  }
```

Comment:
please can we can we try using middleware instead to avoid validations in the controller.